### PR TITLE
[fix] default database detection from dsn

### PR DESCRIPTION
**Pull request by divine**
Refactored default database function which detects based on db config or tries to detect it from dsn;

Closes: https://github.com/jenssegers/laravel-mongodb/issues/1861

### CODE REVIEW

File: tests/ConnectionTest.php [hunk 1]
```diff
@@ -32,6 +32,15 @@ public function testDb()
         $this->assertInstanceOf(\MongoDB\Client::class, $connection->getMongoClient());
     }
 
+    public function testDsnDb()
+    {
+        $connection = DB::connection('dsn_mongodb_db');
+        $this->assertInstanceOf(\MongoDB\Database::class, $connection->getMongoDB());
+
+        $connection = DB::connection('dsn_mongodb_db');
```
Comment (Smolevich) [inline tests/ConnectionTest.php]: What is the profit to create new instance connection?

File: tests/ConnectionTest.php [hunk 2]
```diff
@@ -32,6 +32,15 @@ public function testDb()
         $this->assertInstanceOf(\MongoDB\Client::class, $connection->getMongoClient());
     }
 
+    public function testDsnDb()
+    {
+        $connection = DB::connection('dsn_mongodb_db');
+        $this->assertInstanceOf(\MongoDB\Database::class, $connection->getMongoDB());
```
Comment (Smolevich) [inline tests/ConnectionTest.php]: Add `\MongoDB\Database` to section use

File: tests/ConnectionTest.php [hunk 3]
```diff
@@ -32,6 +32,15 @@ public function testDb()
...
+
+        $connection = DB::connection('dsn_mongodb_db');
+        $this->assertInstanceOf(\MongoDB\Client::class, $connection->getMongoClient());
```
Comment (Smolevich) [inline tests/ConnectionTest.php]: Leave only `Client::class` please

Comment (divine) [reply tests/ConnectionTest.php]: I've just copied from upper function, see https://github.com/jenssegers/laravel-mongodb/blob/master/tests/ConnectionTest.php#L26

Should we refactor all tests? It's not like me who did that 😉

Comment (Smolevich) [reply tests/ConnectionTest.php]: OK, I suggest refactor tests in another PR

**Comment by derekrprice** [comment]
That worked perfectly, thanks!

**Pull request merged**